### PR TITLE
ENG-11055: Infrastructure for @Rebalance

### PR DESCRIPTION
**Pull request by zeemanhe**
Created infrastructure for new stored proc @Rebalance.

### CODE REVIEW

File: lib/python/voltcli/voltadmin.d/rebalance.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+
+@VOLT.Command(
+    bundles = VOLT.AdminBundle(),
+    options = (
+        VOLT.IntegerOption('-c', '--count', 'hostcount', 'number of hosts in the cluster'),
+        VOLT.StringOption('-t', '--stop', 'stoplist', 'target nodes to be stopped'),
+        VOLT.IntegerOption('-k', '--kfactor', 'kfactor', 'kfactor in cluster'),
+    ),
+    description = 'Rebalance the cluster.',
```
Comment (xinjiacs) [inline lib/python/voltcli/voltadmin.d/rebalance.py]: Adding message like ", not read to use"

File: lib/python/voltcli/voltadmin.d/rebalance.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    hideverb = True
```
Comment (xinjiacs) [inline lib/python/voltcli/voltadmin.d/rebalance.py]: Does this property control whether "rebalance" will be shown or not when "voltadmin -h" is called?

File: lib/python/voltcli/voltadmin.d/rebalance.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        for host in self.hosts_by_id.values():
+            if host.hostname == host_name:
+                return host
+        return None
+
+
+@VOLT.Command(
+    bundles = VOLT.AdminBundle(),
+    options = (
+        VOLT.IntegerOption('-c', '--count', 'hostcount', 'number of hosts in the cluster'),
```
Comment (xinjiacs) [inline lib/python/voltcli/voltadmin.d/rebalance.py]: --hostcount other than '--count'?

"target number of hosts in the cluster"

File: lib/python/voltcli/voltadmin.d/rebalance.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with VoltDB.  If not, see <http://www.gnu.org/licenses/>.
+
+class Host(dict):
```
Comment (xinjiacs) [inline lib/python/voltcli/voltadmin.d/rebalance.py]: Can you refactor out these two classes? It has been used in STOP command as well.

File: src/frontend/org/voltdb/SystemProcedureCatalog.java [hunk 5]
```diff
@@ -179,6 +179,7 @@ Procedure asCatalogProcedure() {
         builder.put("@ExplainProc",             new Config(null,                                           true,  true,  false, 0,    VoltType.INVALID,   false, false, true,  true,      false,    false));
         builder.put("@SendSentinel",            new Config(null,                                           true,  false, false, 0,    VoltType.INVALID,   true,  false, false, true,      false,    false));
         builder.put("@PrepareShutdown",         new Config("org.voltdb.sysprocs.PrepareShutdown",          false, false, false, 0,    VoltType.INVALID,   false, false, true,  true,      false,    true));
+        builder.put("@Rebalance",               new Config(null,                                           true,  false, false, 0,    VoltType.INVALID,   false, false, true,  true,      false,    false));
```
Comment (xinjiacs) [inline src/frontend/org/voltdb/SystemProcedureCatalog.java]: Not sure if it's SP, but we can just leave it here util the REBALANCE is really used.

File: lib/python/voltcli/voltadmin.d/rebalance.py (hunk 4 above)
Comment (zeemanhe) [inline lib/python/voltcli/voltadmin.d/rebalance.py]: sure. will do

File: lib/python/voltcli/voltadmin.d/rebalance.py (hunk 1 above)
Comment (zeemanhe) [inline lib/python/voltcli/voltadmin.d/rebalance.py]: ok

File: lib/python/voltcli/voltadmin.d/rebalance.py (hunk 2 above)
Comment (zeemanhe) [inline lib/python/voltcli/voltadmin.d/rebalance.py]: correct. with hideverb = True, the command will not be list on voltadmin -h

File: lib/python/voltcli/voltadmin.d/rebalance.py (hunk 3 above)
Comment (zeemanhe) [inline lib/python/voltcli/voltadmin.d/rebalance.py]: will change

**Pull request merged**